### PR TITLE
Resolve component version conflict

### LINKED ISSUE #698: elastic-job 依赖jar包升级问题
elastic-job 2.1.5版本，但是依赖的，zk 以及 Curator-* 版本依旧很低，是否有考虑过升级依赖jar包版本？
如果去掉elastic-job  中的zk相关依赖包进行单独进行升级，能否给出些升级参考意见？

### PR DESCRIPTION
**Pull request by Leonidasz**
Fixes #698 #590 #475.

Changes proposed in this pull request:
- Module elastic-job-lite-spring removed dependency of guava.
- Update component verion, package the conflict-prone components into elastic-job-lite-core, and make relocation.
    Internally update curator version to 2.13.0
    Internally update zk version to 3.4.8
    Package curator-* and guava into elastic-job-lite-core, and make relocation.
- Change guava version to 16.0.1  
- Adjust the dependency transitive.

### CODE REVIEW

**Review by haocao — COMMENTED** [review]
<id>elastic-job-lite-shader</id>

File: elastic-job-lite-core/pom.xml [hunk 1]
```diff
@@ -28,20 +28,32 @@
...
         </dependency>
         <dependency>
             <groupId>org.apache.curator</groupId>
             <artifactId>curator-recipes</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.zookeeper</groupId>
+            <artifactId>zookeeper</artifactId>
+            <version>3.4.8</version>
```
Comment (haocao) [inline elastic-job-lite-core/pom.xml]: 3.4.8 extract to variable ${zookeeper.version}

File: elastic-job-lite-spring/pom.xml [hunk 2]
```diff
@@ -28,6 +28,21 @@
...
+            <artifactId>curator-client</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.curator</groupId>
+            <artifactId>curator-recipes</artifactId>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.apache.curator</groupId>
```
Comment (haocao) [inline elastic-job-lite-spring/pom.xml:33]: remove curator-test and curator-framework

File: elastic-job-lite-core/pom.xml (hunk 1 above)
Comment (Leonidasz) [reply elastic-job-lite-core/pom.xml]: Get it~

File: elastic-job-lite-spring/pom.xml (hunk 2 above)
Comment (Leonidasz) [reply elastic-job-lite-spring/pom.xml:33]: Ok!

File: elastic-job-lite-spring/pom.xml [hunk 3]
```diff
@@ -28,6 +28,11 @@
...
             <artifactId>lombok</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.curator</groupId>
+            <artifactId>curator-recipes</artifactId>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.apache.curator</groupId>
             <artifactId>curator-test</artifactId>
```
Comment (haocao) [inline elastic-job-lite-spring/pom.xml:34]: remove curator-test

**Pull request merged**